mine-diamonds/api/README.md

```diff
@@ -0,0 +1,152 @@
+
+The CLI will generate a `routes.ts` file, which exports a function called `RegisterRoutes`. The function takes an Express server as its sole argument. This package exports that function, to be imported by the game backend and passed to Paima Engine to run the server.
+
+## Endpoints
+
+### user_stats
+
+Gets data about specified `wallet` -- their `currentusertokenid`, which is the incremental ID counter for the game-side token representation.
+
+| Parameter | Type      | Description                                 |
+| :-------- | :-------- | :------------------------------------------ |
+| `wallet`  | `string`  | **required**. The user address              |
+
+#### Sample request
+
+```http
+http://localhost:3333/user_stats?wallet=0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266
+```
+
+#### Sample response
+
+```javascript
+{
+    "stats": {
+        "wallet": "0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266",
+        "currentusertokenid": 2
+    }
+}
+```
+
+### user_token_state
+
+Gets data about the game-side token representation of specific `userTokenId` of specified `wallet`.
+
+| Parameter     | Type      | Description                                 |
+| :------------ | :-------- | :------------------------------------------ |
+| `wallet`      | `string`  | **required**. The user address              |
+| `userTokenId` | `integer` | **required**. The token id of user          |
+
+#### Sample request
+
+```http
+http://localhost:3333/user_token_state?wallet=0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266&userTokenId=1
+```
+
+#### Sample response
+
+```javascript
+{
+    "stats": [
+        {
+            "wallet": "0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266",
+            "usertokenid": 1,
+            "amount": 3,
+            "isdiamond": true
+        }
+    ]
+}
+```
+
+### user_asset_state
+
+Gets data about the minted ERC1155 token with specific `userTokenId` of specified `user`.
+
+| Parameter     | Type      | Description                                 |
+| :------------ | :-------- | :------------------------------------------ |
+| `user`        | `string`  | **required**. The user address              |
+| `userTokenId` | `integer` | **required**. The token id of user          |
+
+#### Sample request
+
+```http
+http://localhost:3333/user_asset_state?user=0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266&userTokenId=1
+```
+
+#### Sample response
+
+```javascript
+{
+    "stats": [
+        {
+            "assettokenid": 1,
+            "minter": "0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266",
+            "usertokenid": 1,
+            "amount": 3
+        }
+    ]
+}
+```
+
+### user_valid_minted_assets
+
+Gets array of ERC1155 token of specified `user` that have been minted in a valid way -- meaning the `amount` and `assetTokenId`/`userTokenId` combination matches.  
```

Comment:
@matejos I guess this endpoint also includes transferred and brought tokens (not only minted ones)

Reply:
Perhaps the wording isn't the best, it's supposed to mean `ERC1155 tokens (assets) that have been minted validly` - meaning their `userTokenId` and `amount` matches with the `userTokenId` and `amount` combination on the game side. This endpoint returns such ERC1155 tokens that are valid representations of the game event.